aws-sdk-s3-transfer-manager/src/operation/download_objects/list_objects.rs

```diff
@@ -101,7 +103,14 @@ impl State {
 
 impl ListObjectsPaginator {
     fn new(context: DownloadObjectsContext) -> Self {
-        let prefix = context.state.input.key_prefix.to_owned();
+        let mut prefix = context.state.input.key_prefix.to_owned();
+        let delimiter = context.state.input.delimiter().unwrap_or(DEFAULT_DELIMITER);
+        // S3Express requires that prefix must end with delimiter
+        if let Some(prefix) = prefix.as_mut() {
```

Comment:
Shouldn't we only do this for S3 express buckets (and we have a function to test if a bucket is s3 express don't we)?

Reply:
I thought it won't hurt for s3 buckets as well but this is not the correct solution as prefix may or may not end with delimiter. It will have different output depending upon the bucket structure like if we have something like
```
prefixFile1
prefixFile2
prefix/file3
prefix/file4
```. 
I am looking more into it, maybe we just need to fix it in the benchmark runner.

Reply:
Yeah I'm a little worried about overriding the prefix by default especially since it doesn't have to end in a delimiter (for regular buckets anyway). You _could_ argue this is just bad input and should be fixed by caller (i.e. the runner as you suggested).

Reply:
@aajtodd Thanks, yeah. I also discussed it with Dmitriy and will fix it in the runner.